test/action_controller/json_api/pagination_test.rb

```diff
@@ -0,0 +1,116 @@
+require 'test_helper'
+require 'will_paginate/array'
```

Comment:
Just looking at [will_paginate/array, could you](https://github.com/mislav/will_paginate/blob/master/lib/will_paginate/array.rb)

```ruby
@array = []
options = {}
require 'will_paginate/collection'
page     = options[:page] || 1
per_page = options[:per_page] || WillPaginate.per_page
total    = options[:total_entries] || @array.length

WillPaginate::Collection.create(page, per_page, total) do |pager|
  pager.replace @array[pager.offset, pager.per_page].to_a
end
```